src/vivarium/framework/utilities.py

```diff
@@ -28,20 +29,63 @@ def to_yearly(value: TimeValue, time_step: Timedelta) -> TimeValue:
     return value / (time_step.total_seconds() / (60 * 60 * 24 * 365.0))
 
 
-def rate_to_probability(rate: Sequence[float] | NumberLike) -> NumericArray:
-    # encountered underflow from rate > 30k
-    # for rates greater than 250, exp(-rate) evaluates to 1e-109
-    # beware machine-specific floating point issues
+def rate_to_probability(
+    rate: Sequence[float] | NumberLike, time_scaling_factor: float | int = 1.0
+) -> NumericArray:
+    """Converts a rate to a probability.
 
-    rate = np.array(rate)
-    rate[rate > 250] = 250.0
-    probability: NumericArray = 1 - np.exp(-rate)
+    Parameters
+    ----------
+    rate
+        The rate to convert to a probability.
+    time_scaling_factor
+        The time factor in to scale the rates. This is uusually the time step.
+
+    Returns
+    -------
+        An array of floats representing the probability of the converted rates
+    """
+    # NOTE: The default behavior for randomness streams is to use a rate that is already
+    # scaled to the time step which is why the default time scaling factor is 1.0.
+    probability = rate * time_scaling_factor
+    # Clip to 1.0 if the probability is greater than 1.0.
+    clip = False
+    exceeds_one = probability > 1.0
+    if isinstance(exceeds_one, bool):
+        if exceeds_one:
+            probability = 1.0
+            clip = True
+    else:
+        if exceeds_one.any():
+            probability[exceeds_one] = 1.0
+            clip = True
+    if clip:
+        logger.warning(
+            "The rate to probability conversion resulted in a probability greater than 1.0. "
+            "The probability has been clipped to 1.0 and indicates the rate is too high. "
+        )
```

Comment:
Can't we do this more or less the same way as we had done before?

```suggestion
    probability: NumericArray = np.array(rate) * time_scaling_factor
    exceeds_one = probability > 1.0
    if exceeds_one.any():
        # Clip to 1.0 if the probability is greater than 1.0.
        probability[exceeds_one] = 1.0
        logger.warning(
            "The rate to probability conversion resulted in a probability greater than 1.0. "
            "The probability has been clipped to 1.0 and indicates the rate is too high. "
        )
```

Reply:
Because of the data types rate can be (float vs series for instance) probability > 1.0 will be a bool or an array so I have to do the separate checks here unless there is a better way to write this.

Reply:
Fixed this now